From f4fe0970d5f90aa71ea7cd86dcc82ef31019aa06 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:37:13 -0400
Subject: [ethernet] i40e/i40evf: unhide and enable to one prefena field

Message-id: <1409658042-22689-73-git-send-email-sassmann@redhat.com>
Patchwork-id: 90267
O-Subject: [RHEL7.1 PATCH 072/281] i40e/i40evf: unhide and enable to one prefena field
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jesse Brandeburg <jesse.brandeburg@intel.com>

The PREFENA field in the receive host memory cache (RX-HMC)
must be visible in order to be set to 1 at driver init for
best performance.

Change-ID: I16b0bcd84cf56f4b6c938201ff5e954bee5a1992
Signed-off-by: Jesse Brandeburg <jesse.brandeburg@intel.com>
Acked-by: Shannon Nelson <shannon.nelson@intel.com>
Tested-by: Kavindya Deegala <kavindya.s.deegala@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 0b3aec852b40d47315956487a902cfb513d8c482)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_lan_hmc.c b/drivers/net/ethernet/intel/i40e/i40e_lan_hmc.c
index d5d98fe..5c341ae 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_lan_hmc.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_lan_hmc.c
@@ -747,6 +747,7 @@ static struct i40e_context_ele i40e_hmc_rxq_ce_info[] = {
  { I40E_HMC_STORE(i40e_hmc_obj_rxq, tphdata_ena),  1, 195 },
  { I40E_HMC_STORE(i40e_hmc_obj_rxq, tphhead_ena),  1, 196 },
  { I40E_HMC_STORE(i40e_hmc_obj_rxq, lrxqthresh),   3, 198 },
+ { I40E_HMC_STORE(i40e_hmc_obj_rxq, prefena),      1, 201 },
  { 0 }
 };
 
diff --git a/drivers/net/ethernet/intel/i40e/i40e_lan_hmc.h b/drivers/net/ethernet/intel/i40e/i40e_lan_hmc.h
index 341de92..eb65fe2 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_lan_hmc.h
+++ b/drivers/net/ethernet/intel/i40e/i40e_lan_hmc.h
@@ -56,6 +56,7 @@ struct i40e_hmc_obj_rxq {
  u8  tphdata_ena;
  u8  tphhead_ena;
  u8  lrxqthresh;
+ u8  prefena; /* NOTE: normally must be set to 1 at init */
 };
 
 /* Tx queue context data */
diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index d7ffcd5..345db09 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -3140,6 +3140,8 @@ static void i40e_netpoll(struct net_device *netdev)
  pf->flags &= ~I40E_FLAG_IN_NETPOLL;
 }
 #endif
+ /* set the prefena field to 1 because the manual says to */
+ rx_ctx.prefena = 1;
 
 /**
  * i40e_vsi_control_tx - Start or stop a VSI's rings
-- 
1.7.1